### PR TITLE
Merge branch 'develop' into 'master'

### LINKED ISSUE #2204: Alert user if extract fails due to path length, and LongPathsEnabled != 1
Per https://github.com/lukesampson/scoop-extras/issues/889#issuecomment-381621827 , let’s add a check to https://github.com/lukesampson/scoop/blob/master/libexec/scoop-checkup.ps1 that we are under Windows 10 Version 1607, or later, and if LongPathsEnabled in HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem is not set to 1, alert the user.

### LINKED ISSUE #2270: Better documenting of environment variables
Just clean reinstalled my windows to the last build, and I took the chance to rethink the whole layout.
I permanently and globally moved my Users folder to another partition (D:) and, as soon as my OS is ready, I proceeded to install scoop.

Part of my layout rethinking also was how and where programs installed. At first, I took a look at how scoop use its environment variables, and I saw that I could change many of them. So, I decided that I want users installed programs available for all users, but in the same partition of my users, and globally available tools - development, etc... - in the root of the first partition. As I already used c:\Tools for some other addons and programs, I opted for it. So, that's becomes my setup routine:

After opened Power Shell:
```
[environment]::setEnvironmentVariable('SCOOP','D:\Tools','User')
$env:SCOOP='D:\Tools'
```
That's put user installed apps in a folder available for all users to read from.
Than:
```
[environment]::setEnvironmentVariable('SCOOP_GLOBAL','C:\Tools','User')
$env:SCOOP_GLOBAL='C:\Tools'
```
That tell to scoop to install in C:\Tool all global programs, for a convenient and clean approach.

Finally:
```
iex (new-object net.webclient).downloadstring('https://get.scoop.sh')
```
and all is ready to go!

I think that it is not a real issue, but if environment variables could be better documented, many other users can benefit from them!

### LINKED ISSUE #2454: license property in autoupdate missing in schema.json
From AppVeyor;
```
    [*] wlp-javaee.json: Property 'license' has not been defined and the schema does not allow additional properties.
      [-] wlp-javaee.json 144ms
        Expected 0, but got 1.
        186:                         $validator.Errors.Count | should be 0
        at Invoke-LegacyAssertion, C:\Program Files\WindowsPowerShell\Modules\Pester\4.4.0\Functions\Assertions\Should.ps1: line 188
        at <ScriptBlock>, C:\projects\scoop-ibm\Scoop-Bucket.Tests.ps1: line 186
```

In [`wlp-javaee.json`](https://github.com/se35710/scoop-ibm/blob/master/wlp-javaee.json):

```
    "autoupdate": {
        "url": "https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/$version/wlp-javaee7-$version.zip",
        "license": "https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/downloads/wlp/$version/lafiles/runtime/en.html"
},
```

During updates, the license property is updated as expected.

### LINKED ISSUE #2735: Ask if user wants to review install script before installing packages
As we all know, Scoop buckets are maintained by volunteers. It's great that people put in that effort to help the community. But what if a volunteer's GitHub account gets hacked, because they haven't turned on two-factor authentication for example? The hacker could hide malicious code in a popular Scoop package that way. Or what if someone with bad intentions manages to become a contributor? If I'm not mistaken, there's no code review process and every volunteer can make changes without any form of scrutiny.

Making the contribution process stricter is one way to tackle that, but my feature request focuses on the end user's side. I use a package manager on Arch Linux that lets me install community-maintained packages and it has a feature to review the install script (in Scoop's case: a JSON file) before I continue.

My idea:

- I do `scoop install signal`.
- Scoop asks: "Do you want to review the install file first? [y/N]"
- If I answer yes, it opens in a text editor (vim/nano/less/notepad/whatever). After closing it, Scoop asks: "Still want to install this package? [Y/n]"
- If I answer yes, install continues as normal.

Would you guys be open to adding this check?

### LINKED ISSUE #4530: fix "scoop update -a"
found [this one](https://github.com/ScoopInstaller/Scoop/commit/0d721abc2f200f9d0dda2fa26d7b68710a3a9840) finally merged (THANKS!!!) and was super excited to test it... 

PRO:
* ```scoop help update``` shows already the merged help info:
```powershell
λ  scoop help update
Usage: scoop update <app> [options]

'scoop update' updates Scoop to the latest version.
'scoop update <app>' installs a new version of that app, if there is one.

You can use '*' in place of <app> to update all apps.

Options:
  -f, --force               Force update even when there isn't a newer version
  -g, --global              Update a globally installed app
  -i, --independent         Don't install dependencies automatically
  -k, --no-cache            Don't use the download cache
  -s, --skip                Skip hash validation (use with caution!)
  -q, --quiet               Hide extraneous messages
  -a, --all                 Update all apps (alternative to '*')
  ```
* ```scoop update --all``` works great

CON: ```scoop update -a``` doesn't.. getting

```powershell
λ  scoop update -a
scoop update: Option -a requires an argument.
```

### PR DESCRIPTION
**Pull request by niheaven**
Related PRs:
- #4531 @filmor 
  - ac71fcc @niheaven 
- #4535 @rashil2000 
- #4522 @pratikpc 
- #4550 @niheaven 
- #4528 @niheaven 
- #4532 @MrNuggelz
- #4155 @MrNuggelz Co-authored-by: @rashil2000 
  - #4581 @niheaven 
  - fb496c4 @rashil2000 
- #4543 @rashil2000 Co-authored-by: @niheaven 
  - #4555 @rashil2000 
  - 3c90d1a @rashil2000 
  - 2ec00d5 @rashil2000 
- #4567 @rashil2000 
  - cbe29ed @rashil2000 
- #4570 @niheaven 
  - #4582 @niheaven 
- #4571 @niheaven 
- #3244 @nickbudi 
- #3821 @jfastnacht Co-authored-by: @rasa 
- #4578 @tukanos
- #4579 @rashil2000 

Fixed or closed issues:
- Fix #4530 
- Fix #2454 
- Close #2735
- Close #2204
- Close #2270

### CODE REVIEW

**Comment by rashil2000** [comment]
Regarding https://github.com/ScoopInstaller/Scoop/pull/4528#issuecomment-975775263, I think we should do something about it - otherwise any manifest without URL will fail. See https://github.com/ScoopInstaller/Scoop/issues/4285.

**Comment by niheaven** [comment]
For something like meta manifests that without url, a lots of work to support them should be done, and IMO it is out of work of these PRs.

**Comment by rashil2000** [comment]
Okay, but we should still do something about the `Cannot bind argument to parameter 'URL' because it is null.` error, maybe in a hotfix later.

**Review by rashil2000 — APPROVED** [review]

**Review by issaclin32 — APPROVED** [review]

File: lib/core.ps1 [hunk 1]
```diff
@@ -604,21 +614,63 @@ set args=%args:(=``(%
...
 function get_shim_path() {
-    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"
+    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"
     $shim_version = get_config 'shim' 'default'
     switch ($shim_version) {
         '71' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\71\shim.exe"; Break }
-        'kiennq' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"; Break }
+        'scoopcs' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"; Break }
+        'kiennq' { Break } # for backward compatibility
         'default' { Break }
```
Comment (rasa) [inline lib/core.ps1:674]: It seems there is no way for the user to select the `shimexe\bin\shim.exe` shim?

**Review by rasa — APPROVED** [review]

File: lib/core.ps1 (hunk 1 above)
Comment (rashil2000) [reply lib/core.ps1:674]: `scoop config shim scoopcs` does that.

Comment (rasa) [reply lib/core.ps1:674]: Oh, duh. It's right there, sorry.

**Comment by LazyGeniusMan** [comment]
Not sure if it's related but AppVeyor start to fail after this is merged. [[1]](https://ci.appveyor.com/project/r15ch13/scoopinstaller-extras/builds/42005806) [[2]](https://ci.appveyor.com/project/r15ch13/scoopinstaller-extras/builds/42006413) [[3]](https://ci.appveyor.com/project/r15ch13/scoopinstaller-versions/builds/42005850)
```
Command executed with exception: Exception calling "Validate" with "1" argument(s): "Could not resolve schema reference 'http://scoop.sh/draft/schema#/definitions/license'. Path 'definitions.autoupdateArch.properties.license', line 224, position 28."
```
Is this related to  https://github.com/ScoopInstaller/Scoop/pull/4528 ? or `appveyor.yml` need to be updated?

**Comment by rashil2000** [comment]
Also, `develop` is still ahead of `master` after this merge. Maybe we shouldn't use `Squash and merge` for these merges?

**Comment by niheaven** [comment]
"Merge' is off in this repo 😢 Indeed, 'develop' and 'master' are identical after this merge, and reverse merge from 'master' to 'develop' should be done after such merge. Do it next time or enable 'merge commits' in the repo.

**Comment by rashil2000** [comment]
I think all team members can change the merge settings. This is what I see:

image

**Pull request merged**